### PR TITLE
intelligently add blank lines to json and yaml config files

### PR DESCRIPTION
**Pull request by rubys**
match the look of toml serialization

example output:

```
% ~/git/flyctl/bin/flyctl config show --local
{
  "app": "demo-rubys-11060",
  "primary_region": "iad",
  "console_command": "/rails/bin/rails console",

  "mounts": [
    {
      "source": "data",
      "destination": "/data"
    }
  ],

  "http_service": {
    "internal_port": 3000,
    "force_https": true,
    "auto_stop_machines": true,
    "auto_start_machines": true,
    "min_machines_running": 0,

    "processes": [
      "app"
    ]
  },

  "checks": {
    "status": {
      "port": 3000,
      "type": "http",
      "interval": "10s",
      "timeout": "2s",
      "grace_period": "5s",
      "method": "GET",
      "path": "/up",
      "protocol": "http",
      "tls_skip_verify": false,

      "headers": {
        "X-Forwarded-Proto": "https"
      }
    }
  },

  "vm": [
    {
      "memory": "1gb",
      "cpu_kind": "shared",
      "cpus": 1
    }
  ],

  "statics": [
    {
      "guest_path": "/rails/public",
      "url_prefix": "/",
      "tigris_bucket": ""
    }
  ]
}
```

### CODE REVIEW

**Comment by rubys** [comment]
yaml example:

```
% ~/git/flyctl/bin/flyctl config show --local --yaml
# fly.yaml app configuration file generated for demo-rubys-11060 on 2024-05-21T22:13:01-04:00
#
# See https://fly.io/docs/reference/configuration/ for information about how to use this file.
#

app: demo-rubys-11060
primary_region: iad
console_command: /rails/bin/rails console

mounts:
  - source: data
    destination: /data

http_service:
  internal_port: 3000
  force_https: true
  auto_stop_machines: true
  auto_start_machines: true
  min_machines_running: 0

  processes:
    - app

checks:
  status:
    port: 3000
    type: http
    interval: 10s
    timeout: 2s
    grace_period: 5s
    method: GET
    path: /up
    protocol: http
    tls_skip_verify: false

    headers:
      X-Forwarded-Proto: https

vm:
  - memory: 1gb
    cpu_kind: shared
    cpus: 1

statics:
  - guest_path: /rails/public
    url_prefix: /
    tigris_bucket: ""
```

File: internal/appconfig/serde.go [hunk 1]
```diff
@@ -73,7 +78,42 @@ func (c *Config) WriteTo(w io.Writer, format string) (int64, error) {
...
+			count, err := w.Write([]byte(text + "\n"))
+			if err != nil {
+				return 0, err
+			}
+			total += count
+		}
+
+		if err := scanner.Err(); err != nil {
+			return 0, err
+		}
```
Comment (dangra) [inline internal/appconfig/serde.go:113]: I'd put all this stuff in a separate private function, name it accordingly and even add a simple test case so in case we found bugs in the regex for some edge case, a future ourselves could fix them with certainty of not breaking past fixes.

**Review by dangra — APPROVED** [review]
